src/Symfony/Component/Security/Http/LoginLink/LoginLinkHandler.php

```diff
@@ -97,8 +97,12 @@ public function consumeLoginLink(Request $request): UserInterface
             throw new InvalidLoginLinkException('User not found.', 0, $exception);
         }
 
-        $hash = $request->get('hash');
-        $expires = $request->get('expires');
+        if (!$hash = $request->get('hash')) {
+            throw new InvalidLoginLinkException('Missing "hash" parameter.');
+        }
+        if (!$expires = $request->get('expires')) {
+            throw new InvalidLoginLinkException('Missing "expires" parameter.');
```

Comment:
Other authenticators throw a `BadRequestHttpException` in such cases, I would do the same here. See e.g. https://github.com/symfony/symfony/blob/c79d4ab7e787eb27d83581ccb8e15a4e11e88df3/src/Symfony/Component/Security/Http/Authenticator/JsonLoginAuthenticator.php#L161

Reply:
Hmm `LoginLinkHandlerInterface::consumeLoginLink` is supposed to throw `InvalidLoginLinkException`: https://github.com/symfony/symfony/blob/f67ff7640624a53b3f0d096a2d7dbba5a5369001/src/Symfony/Component/Security/Http/LoginLink/LoginLinkHandlerInterface.php#L32-L34

The `LoginLinkAuthenticator` then converts it to `InvalidLoginLinkAuthenticationException` which triggers its `onAuthenticationFailure` method to be called.

Are you sure we should bypass this behavior?

Reply:
Thinking twice about it, I think it's safer to go with your approach.